src/app.rs

```diff
@@ -154,6 +157,49 @@ impl App {
     pub fn get_pose_estimate(&self) -> rosrust_msg::geometry_msgs::Transform {
         transformation::iso2d_to_ros(&self.pose_estimate)
     }
+
+    pub fn deactivate_image_subs(&mut self) {
+        for sub in self.listeners.images.iter_mut() {
+            sub.deactivate();
+        }
+    }
+
+    pub fn activate_next_image_sub(&mut self) {
+        let n_subs = &self.listeners.images.len();
+        for (i, sub) in self.listeners.images.iter_mut().enumerate() {
+            if sub.is_active() {
+                sub.deactivate();
+                if i < n_subs - 1 {
+                    let _ = &self.listeners.images[i + 1].activate();
+                } else {
+                    let _ = &self.listeners.images[0].activate();
+                }
+                return;
+            }
+        }
+        if n_subs > &0 {
+            let _ = &self.listeners.images[0].activate();
+        }
+    }
+
+    pub fn draw_image<B>(&mut self, f: &mut Frame<B>)
+    where
+        B: Backend,
+    {
+        let chunks = Layout::default()
+            .constraints([Constraint::Percentage(100)].as_ref())
+            .split(f.size());
+        for image_sub in &self.listeners.images {
```

Comment:
Could you not just store the currently active image listener?
It does not really matter but it feels a bit weird this way

Reply:
i didnt want to store it in an extra attributes, the subscribers themselves 'know' when they are active.